src/app/nodetool/qgsnodeeditor.cpp

```diff
@@ -227,10 +228,32 @@ void QgsNodeEditor::updateNodeSelection()
 {
   mSelectedFeature->blockSignals( true );
   mSelectedFeature->deselectAllVertexes();
-  Q_FOREACH ( const QModelIndex& index, mTableWidget->selectionModel()->selectedRows() )
+  Q_FOREACH( const QModelIndex& index, mTableWidget->selectionModel()->selectedRows() )
   {
     int nodeIdx = mTableWidget->item( index.row(), 0 )->data( Qt::DisplayRole ).toInt();
     mSelectedFeature->selectVertex( nodeIdx );
   }
   mSelectedFeature->blockSignals( false );
 }
+
+void QgsNodeEditor::zoomToNode( QTableWidgetItem *item )
+{
+  double x, y;
+  x = mTableWidget->item( item->row(), 1 )->data( Qt::DisplayRole ).toDouble();
+  y = mTableWidget->item( item->row(), 2 )->data( Qt::DisplayRole ).toDouble();
+
+  QgsRectangle ext = mCanvas->extent();
+  QgsPoint center = ext.center();
+
+  QgsPoint newCenter( x, y );
+
+  QgsCoordinateTransform t( mLayer->crs(), mCanvas->mapSettings().destinationCrs() );
+  QgsPoint tCenter = t.transform( newCenter );
```

Comment:
better to use QgsMapCanvas::setCenter( tCenter ) here instead of lines 253-257? would simplify things a bit.